### ISSUE TITLE
Fiber expects a function

### ISSUE THREAD

**Issue by jpaulm** [laverdet/node-fibers#213]
I am getting a new message after reinstalling node-fibers:  Fiber expects a function

The code is:
```
x.fiber = new Fiber(x.func);
``` 
where x is my Process class, in which `func` is a function, and `fiber` will be a Fiber.  

I put a `console.log(x);` in front of this statement, and got the following on my console:

```
{ name: 'reader',
  func: { reader: [Function] },
  fiber: null,
  inports: [ [ 'reader.FILE', [Object] ] ],
  outports: [ [ 'reader.OUT', [Object] ] ],
  status: 'N',
  ownedIPs: 0,
  cbpending: false,
  yielded: false,
  data: null }
C:\Users\Paul\Documents\GitHub\FBP-Repos\jsfbp-master\index.js:444
  }).run();
     ^
TypeError: Fiber expects a function
    at TypeError (native)
    at run2 (C:\Users\Paul\Documents\GitHub\FBP-Repos\jsfbp-master\index.js:484:
17)
```

It worked fine before - could the message be due to a change in fibers?  TIA

PS @tlrobinson have you run into this?

**Comment by laverdet** [comment]
The error is in your code.

**Comment by jpaulm** [comment]
Thanks, Marcel, I was beginning to suspect that!

**Comment by tlrobinson** [comment]
@jpaulm I agree with @laverdet this is likely an issue with our code, let's move the discussion back there.

**Comment by jpaulm** [comment]
@tlrobinson fbptest2 is now working, now that I have merged your fork, for which thanks again!  I am not quite sure what happened the first time I tried it, but I suspect that `x.fiber = new Fiber(x.func);` saw a `reader` which had some kind of problem, and so decided that it wasn't a valid function.  I only got the error messages I posted though!

Since @laverdet has closed this, I guess it just stays closed!  :-) 

Thanks
